nox/manifest.py

```diff
@@ -184,6 +184,21 @@ def make_session(
             func.should_warn[WARN_PYTHONS_IGNORED] = func.python
             func.python = False
 
+        if self._config.extra_pythons:
+            # if extra python is provided, expand the func.python list to
+            # include additional python interpreters
+            extra_pythons = self._config.extra_pythons  # type: List[str]
+            if isinstance(func.python, (list, tuple, set)):
+                func.python = list({*func.python, *extra_pythons})
+            elif not multi and func.python:
+                # if this is multi, but there is only a single interpreter, it
+                # is the reentrant case. The extra_python interpreter shouldn't
+                # be added in that case. If func.python is False, the session
+                # has no backend; treat None and the empty string equivalently.
+                # Otherwise, add the extra specified python.
```

Comment:
Thank you for incorporating the earlier change :)

I just realized that we should not treat `False` and `None` the same here. Unlike False, None does not imply that there is no backend. By default, None will just get us a virtualenv with the same interpreter that Nox runs on. In that case, there's no reason we shouldn't add the extra Python.

```suggestion
            elif not multi and func.python is not False:
                # if this is multi, but there is only a single interpreter, it
                # is the reentrant case. The extra_python interpreter shouldn't
                # be added in that case. If func.python is False, the session
                # has no backend. Otherwise, add the extra specified python.
```